public/model/weather-data-store.js

```diff
@@ -7,22 +7,107 @@ var P = require('pjs').P;
 var d3 = require('d3');
 var SelectionEvents = require('../event-emitters/selection-events.js');
 
-var FakeOvipositionDataStore = P({
+// TODO(jetpack): we should have comments with exact pointers into these papers for where we get our
+// models, along with explanations for how and why we deviate from them...
+
+// See the papers "Vectorial Capacity of Aedes aegypti: Effects of Temperature and Implications for
+// Global Dengue Epidemic Potential" by J. Liu-Helmersson et al [1] and "Impact of human mobility on
+// the emergence of dengue epidemics in Pakistan" by A. Wesolowski et al [2] for the source of the
+// oviposition and mosquito prevalence models.
+//
+// [1] http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0089783
+// [2] http://www.pnas.org/content/112/38/11887.abstract
+
+var OvipositionModel = P({
   init: function(weather_data_store) {
     this.weather_data_store = weather_data_store;
+    // `temp_to_oviposition` ranges from -Infinity to ~8.8. Negative values are treated as 0. We map
+    // this linearly to a green, yellow, orange color gradient.
+    this.oviposition_to_color = d3.scale.linear().domain([0, 4.4, 8.8]).clamp(true)
+      .range(['#b4d642', '#f8cd4e', '#ff7f47']);
+  },
+
+  temp_to_oviposition: function(x) {
```

Comment:
This seems like a pure function, and a good candidate for unit testing. Care to add a basic one? Maybe test monotonicity, and that output values are in a sane range.